app/controllers/pieces_controller.rb

```diff
@@ -16,12 +17,18 @@ def update
     	@color = 'black'
     end
 
+    if @last_move == 0
```

Comment:
again use 

Piece::COLORS[:black] == @last_move

alternatively you could implement a pair of methods in the pieces model and use those throughout the app.

def is_black?
  self.color == Piece::COLORS[:black]
end

def is_white?
  !is_black?
end

Just a suggestion